spec.emu

```diff
@@ -354,39 +357,26 @@ contributors: Ujjwal Sharma, Younies Mahmoud
               1. Let _parts_ be ! PartitionNumberPattern(_nf_, 𝔽(_value_)).
               1. For each Record { [[Type]], [[Value]] } _part_ in _parts_, do
                 1. Append the new Record { [[Type]]: _part_.[[Type]], [[Value]]: _part_.[[Value]], [[Unit]]: _numberFormatUnit_ } to the end of _list_.
-              1. Append _list_ to the end of _result_.
+              1. Set _unitPlaceables_.[[&lt; _numberFormatUnit_ &gt;]] to _list_.
+              1. Append _unit_ to _unitPlaceholders_.
         1. Let _lfOpts_ be OrdinaryObjectCreate(*null*).
         1. Perform ! CreateDataPropertyOrThrow(_lfOpts_, *"type"*, *"unit"*).
         1. Let _listStyle_ be _durationFormat_.[[Style]].
         1. If _listStyle_ is *"digital"*, then
           1. Set _listStyle_ to *"short"*.
         1. Perform ! CreateDataPropertyOrThrow(_lfOpts_, *"style"*, _listStyle_).
         1. Let _lf_ be ! Construct(%ListFormat%, &laquo; _durationFormat_.[[Locale]], _lfOpts_ &raquo;).
-        1. Set _result_ to ! CreatePartsFromList(_lf_, _result_).
-        1. Let _flattened_ be a new empty List.
-        1. For each Record { [[Type]], [[Value]] } _part_ in _result_, do
-          1. If _part_.[[Type]] is *"literal"*, then
-            1. Append _part_ to the end of _flattened_.
-          1. Else,
-            1. Let _value_ be _part_.[[Value]].
-            1. If Type(_value_) is Record { [[Type]], [[Value]], [[Unit]] }, then
-              1. Append _value_ to the end of _flattened_.
-            1. Else,
-              1. Assert: Type(_value_) is List.
-              1. For each Record _element_ in _value_, do
-                1. Append _element_ to the end of _flattened_.
-        1. Return _flattened_.
+        1. Let _placeholders_ be a new empty List.
+        1. For each String _unit_ of _unitPlaceholders_, do
+          1. Let _placeholder_ be the string-concatenation of *"{"*, _unit_, and *"}"*.
+          1. Append _placeholder_ to _placeholders_.
+        1. Let _formatted_ be ! FormatList(_lf_, _placeholders_).
```

Comment:
so... if I have the following code
```js
const duration = {
  days: 2,
  hours: 7,
  minutes: 8,
  seconds: 9,
  milliseconds: 123,
  microseconds: 456,
  nanoseconds: 789,
};
let df = new Intl.DurationFormat('en', { style: "digital" });
let str = df.format(duration);
let out = df.formatToParts(duration);
```
what should be _placeholders_ here?
and what should be _formatted_ here? 
and what should be _unitPlaceables_ here?